libraries/GCS_MAVLink/GCS_FTP.cpp

```diff
@@ -47,9 +47,13 @@ bool GCS_MAVLINK::ftp_init(void) {
     if (ftp.requests == nullptr || ftp.requests->get_size() == 0) {
         goto failed;
     }
-
+    #if CONFIG_HAL_BOARD == HAL_BOARD_ESP32
```

Comment:
should just be an ifndef, then add to board.h

Reply:
I was unaware that chibios had an internal "extra bit" that it adds to each thread created through Scheduler::thread_create ... i've done equivalent in the esp32 HAL now.